blueoil/datasets/pascalvoc_custom.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2020 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+import os
+import os.path
+
+from blueoil.datasets.pascalvoc_base import PascalvocBase
+
+
+class PascalVOCCustom(PascalvocBase):
+    classes = []
+    available_subsets = []
+    extend_dir = ""
+
+    def __init__(
+            self,
+            *args,
+            **kwargs
+    ):
+        PascalVOCCustom.classes = kwargs['classes']
+        PascalVOCCustom.available_subsets = kwargs['available_subsets']
+        PascalVOCCustom.extend_dir = kwargs['extend_dir']
+        super().__init__(
+            *args,
+            **kwargs,
+        )
+
+    @property
+    def num_max_boxes(self):
+        # Took from pascalvoc_2007.py
+        if self.skip_difficult:
+            return 37
+        else:
+            return 42
+
+    def _annotation_file_from_image_id(self, image_id):
+        annotation_file = os.path.join(self.annotations_dir, "{:06d}.xml".format(image_id))
+        return annotation_file
+
+    def _image_file_from_image_id(self, image_id):
+        """Return image file name of a image."""
+        return os.path.join(self.jpegimages_dir, "{:06d}.jpg".format(image_id))
```

Comment:
Maybe these methods need to be updated after https://github.com/blue-oil/blueoil/pull/1072 is merged. `image_id` will be changed from int to str.

Reply:
Thank you for raising this. I will keep aware and update this part again after #1072 merged

Reply:
Sorry #1072 is already merged. So I fixed this.